File: backend/Models/Data/messages/message.py
# coding=utf-8

from datetime import datetime
from sqlalchemy import Column, String, Integer, Float, DateTime, ForeignKey, Boolean
from sqlalchemy.orm import relationship
from Models.entity import Entity
from Configuration.config import DB_SESSION, ENGINE, BASE
from marshmallow import Schema, fields
import random, string

class MessageTable(Entity, BASE):
    __tablename__ = 'messages'

    id = Column(Integer, primary_key=True)
    conversation_id = Column(Integer)
    sender_id = Column(Integer)
    body = Column(String)
    read = Column(Boolean)

    def __init__(self, sender_id, conversation_id, body, created_by):
        Entity.__init__(self, created_by)
        self.sender_id = sender_id
        self.conversation_id = conversation_id
        self.body = body
        self.read = False

class MessageTableSchema(Schema):
    id = fields.Number()
    sender_id = fields.Number()
    conversation_id = fields.Number()
    body = fields.Str()
    read = fields.Boolean()

    created_at = fields.DateTime()
    updated_at = fields.DateTime()
    last_updated_by = fields.Str()

File: frontend/src/Helpers/Messages/Classes/Message.ts
import { IMessage } from '../Interfaces/IMessage';

export class Message implements IMessage {
    private id: number;
    private conversationId: number;
    private senderId: number;
    private body: string;
    private read: boolean;
    private createdAt: Date;
    private updatedAt: Date;

    public get Body(): string {
        return this.body;
    }
    public set Body (value: string) {
        this.body = value;
    }

    public get Id(): number {
        return this.id;
    }
    public set Id (value: number) {
        this.id = value;
    }
    public get ConversationId(): number {
        return this.conversationId;
    }
    public set ConversationId (value: number) {
        this.conversationId = value;
    }
    public get SenderId(): number {
        return this.senderId;
    }
    public set SenderId (value: number) {
        this.senderId = value;
    }

    public get Read(): boolean {
        return this.read;
    }
    public set Read (value: boolean) {
        this.read = value;
    }

    public get CreatedAt(): Date {
        return this.createdAt;
    }
    public set CreatedAt (value: Date) {
        this.createdAt = value;
    }
    public get UpdatedAt(): Date {
        return this.updatedAt;
    }
    public set UpdatedAt (value: Date) {
        this.updatedAt = value;
    }

    constructor(id: number, conversationId:number, senderId:number, body:string, read: boolean, createdAt: Date, updatedAt: Date) {
        this.id = id;
        this.conversationId = conversationId;
        this.senderId = senderId;
        this.body = body;
        this.read = read;
        this.createdAt = createdAt;
        this.updatedAt = updatedAt;
    }
    
}